graylog2-web-interface/src/components/inputs/InputsList.jsx

```diff
@@ -55,7 +65,12 @@ const InputsList = createReactClass({
       localInputs = localInputs.filter(input => input.node === this.props.node.node_id);
     }
 
-    this.setState({ globalInputs: globalInputs, localInputs: localInputs });
+    this.setState({
+      globalInputs: globalInputs,
+      filteredGlobalInputs: globalInputs,
+      localInputs: localInputs,
+      filteredLocalInputs: localInputs,
+    });
```

Comment:
We also need to update the filtered inputs when props change. For instance, when a new input is added, we should keep the filter or at least reset it. Right now the filter seems to be in place, but the list is not filtered.

Reply:
Sorry, I meant when the state propagated from the store changes :)